src/Language/PureScript/Sugar/Names.hs

```diff
@@ -529,6 +529,7 @@ resolveImport currentModule importModule exps imps impQual =
   isHidden hidden ref@(TypeRef _ _) =
     let
       checkTypeRef _ True _ = True
+      checkTypeRef r acc (PositionedDeclarationRef _ _ h) = checkTypeRef r acc h
```

Comment:
@paf31 this not existing was the cause of my woes, previously `checkedRefs` below was deleting `PositionedDeclarationRef` from the AST which meant this case wasn't needed...